javacord-core/src/main/java/org/javacord/core/util/handler/interaction/InteractionCreateHandler.java

```diff
@@ -64,7 +65,9 @@ public void handle(JsonNode packet) {
 
             // Check if this interaction comes from a guild or a DM
             if (packet.hasNonNull("guild_id")) {
-                channel = api.getTextChannelById(channelId).orElse(null);
+                channel = api.getServerChannelById(channelId)
+                        .flatMap(ServerChannel::asTextableRegularServerChannel)
+                        .orElse(null);
```

Comment:
Makes it easier to debug next time
```suggestion
                        .orElseThrow(AssertionError::new);
```